Wikipedia/Code/WKWebViewWithSettableInputViews.swift

```diff
@@ -89,24 +89,53 @@ class WKWebViewWithSettableInputViews: WKWebView {
     }
 }
 
-private typealias ClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
-private typealias BlockType =  @convention(block) (Any, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
-private extension WKWebViewWithSettableInputViews {
-    private func setKeyboardRequiresUserInteraction(_ value: Bool) {
+//https://stackoverflow.com/questions/32449870/programmatically-focus-on-a-form-in-a-webview-wkwebview
+
+typealias OldClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Any?) -> Void
+typealias NewClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
+
+extension WKWebViewWithSettableInputViews {
+
+    func setKeyboardRequiresUserInteraction( _ value: Bool) {
         guard let WKContentView: AnyClass = NSClassFromString("WKContentView") else {
-            DDLogError("Could not get class")
+            print("keyboardDisplayRequiresUserAction extension: Cannot find the WKContentView class")
             return
         }
-        let sel = sel_getUid("_startAssistingNode:userIsInteracting:blurPreviousNode:changingActivityState:userObject:")
-        guard let method = class_getInstanceMethod(WKContentView, sel) else {
-            DDLogError("Could not get method")
-            return
+        // For iOS 11.3, *
+        let sel_11_3: Selector = sel_getUid("_startAssistingNode:userIsInteracting:blurPreviousNode:changingActivityState:userObject:")
+        // For iOS 12.2, *
+        let sel_12_2: Selector = sel_getUid("_elementDidFocus:userIsInteracting:blurPreviousNode:changingActivityState:userObject:")
+        // For iOS 13.0, *
+        let sel_13_0: Selector = sel_getUid("_elementDidFocus:userIsInteracting:blurPreviousNode:activityStateChanges:userObject:")
+
+        if let method = class_getInstanceMethod(WKContentView, sel_11_3) {
+            let originalImp: IMP = method_getImplementation(method)
+            let original: NewClosureType = unsafeBitCast(originalImp, to: NewClosureType.self)
+            let block : @convention(block) (Any, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void = { (me, arg0, arg1, arg2, arg3, arg4) in
+                original(me, sel_11_3, arg0, !value, arg2, arg3, arg4)
+            }
+            let imp: IMP = imp_implementationWithBlock(block)
+            method_setImplementation(method, imp)
```

Comment:
would adding a `return` after these `method_setImplementation` calls and reversing the order to 13.0, 12.2, 11.3 be a bit more efficient?

Reply:
It's possible multiple are implemented and return-ing would shortcut that, but I haven't checked. Given this is only called once and there doesn't seem to be any performance degradation, I'm ok with leaving it as is